ext/standard/mail.c

```diff
@@ -557,24 +562,39 @@ PHPAPI int php_mail(const char *to, const char *subject, const char *message, co
 			fprintf(sendmail, "%s%s", hdr, line_sep);
 		}
 		fprintf(sendmail, "%s%s%s", line_sep, message, line_sep);
+#ifdef PHP_WIN32
 		ret = pclose(sendmail);
 
 #if PHP_SIGCHILD
 		if (sig_handler) {
 			signal(SIGCHLD, sig_handler);
 		}
 #endif
-
-#ifdef PHP_WIN32
-		if (ret == -1)
 #else
+		wstatus = pclose(sendmail);
+#if PHP_SIGCHILD
+		if (sig_handler) {
+			signal(SIGCHLD, sig_handler);
+		}
+#endif
+		/* Determine the wait(2) exit status */
+		if (wstatus == -1) {
+			MAIL_RET(0);
+		} else if (WIFSIGNALED(wstatus)) {
+			MAIL_RET(0);
+		} else if (!WIFEXITED(wstatus)) {
+			MAIL_RET(0);
+		} else {
+			ret = WEXITSTATUS(wstatus);
+		}
+#endif
+
 #if defined(EX_TEMPFAIL)
 		if ((ret != EX_OK)&&(ret != EX_TEMPFAIL))
 #elif defined(EX_OK)
 		if (ret != EX_OK)
 #else
```

Comment:
Is it intentional that this is not also running for Windows? The commit message is not entirely clear on that and I lack the Windows expertise.

Reply:
Windows use c client so it doesn't spawn external process.

Reply:
I need to actually re-read it how it works - we have got a smtp implementation but seems like it's using popen_ex in some cases.

Reply:
From my read of that documentation, the return value is -1, if `_pclose` fails, otherwise it is the return value of the process. So if `_pclose` returns something other than 0, we should also fail.